File: stack/python/b10773.py
import sys

note = [0] * 100000
N = int(input())
idx = 0
for i in range(N):
    temp = int(sys.stdin.readline())
    if temp:
        note[idx] = temp
        idx += 1
        continue
    idx -= 1
    note[idx] = 0

total = 0
for i in range(idx):
    total += note[i]

print(total)

File: stack/cpp/b10773.cpp
#include <vector>
#include <iostream>
int main() {
	std::vector<int> note;
	note.reserve(100000);
	
	int N;
	std::cin >> N;
	for (int i = 0; i < N; ++i) {
		int temp;
		std::cin >> temp;
		if (temp) {
			note.push_back(temp);
			continue;
		}
		note.pop_back();
	}

	int total = 0;
	for (int i = 0; i < note.size(); ++i) {
		total += note[i];
	}
	std::cout << total << std::endl;
}